lib/hawkular/operations/operations_api.rb

```diff
@@ -335,7 +354,6 @@ def invoke_operation_helper(operation_payload, operation_name = nil, binary_cont
     end
 
     def check_pre_conditions(hash = {}, params = [], &callback)
-      fail 'Handshake with server has not been done.' unless @ws.open?
```

Comment:
Because the client tries to connect again before doing any operation, so we don't need to check if we are connected or not.